### PR TITLE
IP Assignement: match clusterip family in case of single stack clusterip

### PR DESCRIPTION
**Pull request by fedepaol**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind cleanup
> /kind feature
> /kind design
> /kind flake
> /kind failing
> /kind documentation
> /kind regression

**What this PR does / why we need it**:

There is no point in adding a dual stack loadbalancer ip if the cluster ip is single family.

Additionally, when they do not match we feel free to clean the service and reassign:
https://github.com/metallb/metallb/blob/v0.14.9/controller/service.go#L109

which might cause an active service to change its ip.

Fixes https://github.com/metallb/metallb/issues/2724 and https://github.com/metallb/metallb/issues/2723

**Special notes for your reviewer**:

**Release note**:

```release-note
Even if the service is prefer dual stack and dual stack pools are available, provide loadbalancer ips that match address family (single, dual) of the service's cluster ip.
```

### CODE REVIEW

**Comment by fedepaol** [comment]
cc @mquhuy

File: controller/controller_test.go [hunk 1]
```diff
@@ -846,6 +846,63 @@ func TestControllerMutation(t *testing.T) {
...
+			in: &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						AnnotationAddressPool: "pool5",
+					},
+				},
+				Spec: v1.ServiceSpec{
+					Type:           "LoadBalancer",
+					ClusterIPs:     []string{"1.2.3.4"},
+					IPFamilyPolicy: &IPFamilyPolicyRequireDualStack,
```
[the remark this answers is not in the document]
Comment (fedepaol) [reply controller/controller_test.go:888]: Yep, I considered this to be a corner case when I wrote the test. I  can remove the test as we would never get here (because the service won't get cluisterips in first instance and we skip), or even make a stronger reject when we process the service, or leave this here as a safe net.

[the remark this answers is not in the document]
Comment (oribon) [reply controller/controller_test.go:888]: if it's not too big of a change to have an explicit reject I think it'd be better

[the remark this answers is not in the document]
Comment (fedepaol) [reply controller/controller_test.go:888]: done

**Review by oribon — APPROVED** [review]
lgtm

**Comment by kvaps** [comment]
Hello, is there any ETA about releasing this fix?
